### ISSUE TITLE
Decompiler renders UTF-8 literals as allocating byte arrays

### ISSUE THREAD

**Issue by richlander** [richlander/dotnet-inspect#4225]
## Finding

RVA-backed UTF-8 string literals are raised to `new byte[] { ... }` rather than C# UTF-8 literals (`"..."u8`). The output is valid and content-equivalent, but recompilation changes the static-data lowering and can introduce per-call array allocation.

## Real-artifact witnesses

`DotnetInspector.Services.MetadataFieldCache::WriteField`:

```csharp
// Authored
Write(buf, ": "u8);
Write(buf, value);
Write(buf, "\n"u8);
```

```csharp
// Decompiled Full on current main
Write(buf, new byte[] { 58, 32 });
Write(buf, value);
Write(buf, new byte[] { 10 });
```

The same cluster contains five confirmed methodology-v3 frontier rows:

- `MetadataFieldCache::WriteField`
- `MetadataFieldCache::WriteArray`
- `PackageIndexCache::WriteArray`
- `MySqlConnector.ColumnReaders.TextDoubleColumnReader::ReadValue`
- `MySqlConnector.ColumnReaders.TextFloatColumnReader::ReadValue`

Four generic-syntax rows are additional authored candidates, for a likely nine-row population pending exact product probes.

Close negative: the `Write(ReadOnlySpan<byte>)` consumer is represented correctly; the loss occurs when `RvaSpanPass`/`ArrayLiteral` chooses the source spelling.

## Relationship to prior work

#647 and #902 correctly recovered `<PrivateImplementationDetails>` RVA data instead of dropping it or emitting invalid C#. This is the fidelity/allocation follow-up: when metadata and target typing prove a compiler UTF-8 literal lowering, preserve the `u8` spelling rather than forcing an array literal.

## Reproduction

```bash
dotnet run --project tools/DecompilerHarness -c Release -- \
  ~/.nuget/packages/dotnet-inspect.any/0.14.0/tools/net10.0/any/DotnetInspector.Services.dll \
  --dump 'DotnetInspector.Services.MetadataFieldCache::WriteField'
```

## Done signal

- Proven UTF-8 literal static data renders as a `u8` literal and recompiles to the original static-data field/token shape.
- Embedded quotes, escapes, non-ASCII UTF-8, and empty data have canaries.
- Ordinary byte-array initializers and non-UTF-8 blobs remain `new byte[]`.
- The five confirmed frontier rows no longer report operand differences.

Discovered from #3079's first methodology-v3 EVIL run.
